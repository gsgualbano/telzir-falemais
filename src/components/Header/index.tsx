import React from 'react';

import { Animated } from 'react-animated-css';

import {
  Container,
  Text,
  Title,
  Icon,
  Logo,
} from './styles';

/**
 * Cabeçalho da página
 */
const Header = (props: {}) => (
  <Container>
    <Animated animationIn="bounceIn" animationOut="fadeOut" isVisible>
      <Text>TELZIR</Text>
      <Logo>
        <Title>FaleMais</Title>
        <Icon />
      </Logo>
    </Animated>
  </Container>
);

export default Header;
