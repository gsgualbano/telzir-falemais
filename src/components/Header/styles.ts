import styled from 'styled-components';
import TelefoneLogo from '../../assets/logo.svg';

export const Container = styled.div`
  color: #ffffff;
  padding: 20px 10px;
`;

export const Logo = styled.div`
  display: flex;
`;

export const Text = styled.p`
  margin: 5px 0;
  font-size: 36px;
  font-weight: 600;
  letter-spacing: 0.15em;
`;

export const Title = styled.p`
  margin: 5px 0;
  font-family: 'Lobster', sans-serif;
  font-size: 60px;
  font-weight: 600;
`;

export const Icon = styled.img.attrs({
  src: TelefoneLogo,
  alt: 'telefone logo',
})`
  margin-left: 10px;
  width: 60px;
`;
