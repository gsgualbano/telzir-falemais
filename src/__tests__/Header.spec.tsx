import React from 'react';
import Header from '../components/Header';

import { shallow } from '../enzyme';

describe('<Header />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  it('shows correct data', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.contains('TELZIR')).toBeTruthy();
    expect(wrapper.contains('FaleMais')).toBeTruthy();
  });
});
