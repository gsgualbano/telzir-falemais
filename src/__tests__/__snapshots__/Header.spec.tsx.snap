// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<styled.div>
  <Animated
    animateOnMount={true}
    animationIn="bounceIn"
    animationInDelay={0}
    animationInDuration={1000}
    animationOut="fadeOut"
    animationOutDelay={0}
    animationOutDuration={1000}
    className=""
    isVisible={true}
    style={Object {}}
  >
    <styled.p>
      TELZIR
    </styled.p>
    <styled.div>
      <styled.p>
        FaleMais
      </styled.p>
      <styled.img />
    </styled.div>
  </Animated>
</styled.div>
`;
